### ISSUE TITLE
not appropiate handling of embedded certificates (spec cert) when missing SAN and hostnames mismatch

### ISSUE THREAD

**Issue by alonsocamaro** [F5Networks/k8s-bigip-ctlr#3268]
#### Setup Details
 CIS Version : 2.15.1      
 
#### Description  

Note: this issue has been found while doing internal tests, not in a customer environment
 
When using embedded certificates (spec cert) without SANs the following behaviour is found, when using two routes with different FQDNs:

- if the CN of the two Routes match, it works as expected. Even the CIS warns that the Routes cannot be verified showing the following error message:

```
2024/02/02 10:51:44 [DEBUG] Error: SAN is empty on the certificate. So skipping Hostname validation on cert
```
and the two routes works as expected:
```
$ curl -v -k https://nginx-app1-edge.xxx.f5-udf.com 2>&1 | egrep "subject:|OK"
*  subject: C=CH; O=F5CisMulticluster; CN=nginx-app1-edge.xxx.f5-udf.com
< HTTP/1.1 200 OK
[cloud-user@ocp-provisioner issue-mismatch-nosan-matching]$ curl -v -k https://nginx-app1.xxx.f5-udf.com 2>&1 | egrep "subject:|OK"
*  subject: C=CH; O=F5CisMulticluster; CN=nginx-app1.xxx.f5-udf.com
< HTTP/1.1 200 OK
```
- on the other hand, if one of the Routes additionally has a CN that doesn´t match the Route host name, the following behaviour is found:

Applying the route that doesn´t have a matching CN name (and no SAN) the Route is Admitted:

```
[cloud-user@ocp-provisioner issue-mismatch-nosan-onenomatching]$ curl -v -k https://nginx-app1.xxx.f5-udf.com 2>&1 | egrep "subject:|OK"
*  subject: C=CH; O=F5CisMulticluster; CN=nginx-app1-passthrough.xxx.f5-udf.com
< HTTP/1.1 200 OK
 ```

Then I apply a second Route with a matching CN name (and no SAN either) the Route is Admitted too
 
```
[cloud-user@ocp-provisioner issue-mismatch-nosan-onenomatching]$ oc apply -f nginx-app1-edge.route.yaml 
route.route.openshift.io/nginx-app1-edge-xxx created
[cloud-user@ocp-provisioner issue-mismatch-nosan-onenomatching]$ curl -v -k https://nginx-app1-edge.xxx.f5-udf.com 2>&1 | egrep "subject:|OK"
*  subject: C=CH; O=F5CisMulticluster; CN=nginx-app1-edge.xxx.f5-udf.com
< HTTP/1.1 200 OK
```
but testing the first Route again, the one with the unmatching CN, it can be found that the certificate has been replaced with the certificate of the second Route:

```
[cloud-user@ocp-provisioner issue-mismatch-nosan-onenomatching]$ curl -v -k https://nginx-app1.xxx.f5-udf.com 2>&1 | egrep "subject:|OK"
*  subject: C=CH; O=F5CisMulticluster; CN=nginx-app1-edge.xxx.f5-udf.com
< HTTP/1.1 200 OK
```
#### Steps To Reproduce

1) Access my personal folder and access the folder CIS-issue-3268

#### Actual & Expected result

Seems that if the bad CN Route (nginx-app1.xxx.f5-udf.com) get its certificate replaced is because it is possible to check that it doesn´t have a good CN. If this is the case, and If possible the bad CN Route should not be Admitted at all. 

The second Route should not in any case modify the first Route, there seems to be a bad logic there.

#### Observations (if any)
Take into account possible wildcard names in the CERTs

**Comment by trinaths** [comment]
Created [CONTCNTR-4551] for internal tracking.

**Comment by trinaths** [comment]
Fixed in 2.16.
